app/main.py

```diff
@@ -1,6 +1,21 @@
+import functools
 from typing import Callable
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_store = {}
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs) -> Callable:
+        key = args + tuple(sorted(kwargs.items()))
+
+        if key in cache_store:
+            print("Getting from cache")
+            return cache_store[key]
+        else:
+            print("Calculating new result")
+            result = func(*args, **kwargs)
+            cache_store[key] = result
+            return result
```

Comment:
```suggestion
        key = (args, tuple(sorted(kwargs.items()), func.__name__,)

        if key in cache_store:
            print("Getting from cache")
        else:
            print("Calculating new result")
            cache_store[key] = func(*args, **kwargs)
        return cache_store[key]
```
`func.__name__`- is needed for that, because 2 different functions can have the same arguments